rasa/core/policies/unexpected_intent_policy.py

```diff
@@ -397,6 +398,37 @@ def compute_label_quantiles_post_training(
         # value specified in the configuration.
         self.label_quantiles = self._compute_label_quantiles(label_id_scores)
 
+    @staticmethod
+    def _get_trackers_for_training(
+        trackers: List[TrackerWithCachedStates],
+    ) -> List[TrackerWithCachedStates]:
+        """Filters out the list of trackers which should not be used for training.
+
+        `UnexpecTEDIntentPolicy` cannot be trained on trackers with:
+        1. `UserUttered` events with no intent.
+        2. `ActionExecuted` events with no action_name.
+
+        Trackers with such events are filtered out.
+
+        Args:
+            trackers: All trackers available for training.
+
+        Returns:
+            Trackers which should be used for training.
+        """
+        trackers_for_training = []
+        for tracker in trackers:
+            tracker_compatible = True
+            for event in tracker.events:
```

Comment:
Why not `tracker.applied_events()` here? Is this because we are assuming training trackers don't don't have things like restarts or event rollbacks?

Reply:
Yeah very very unlikely that training trackers will have those. But even if they do, they will be applied inside the tracker featurizer. So, if anything we are looking at more number of events over here.

Reply:
No strong opinions here TBH. `applied_events` would also work. It didn't cross my mind honestly. 😄

Reply:
Let me change it to applied_events actually. Thanks for pointing it out.